MetaMask/metamask-extension
# Box types update

## Explanation
We are slowly adding more typescript support to the Box component. This PR includes some more additions
- Improves types in the `box.d.ts` file
- Converts `box.stories.js` => `box.stories.tsx` to see linting issues if any occur which will help identify any type issues
- Converts `box.test.js` => `box.test.tsx`
- Adds descriptions to `box.d.ts` file
- Fixes some propTypes in other files that will need to be changed once the Box component is converted to typescript

## Screenshots/Screencaps



### Before

https://user-images.githubusercontent.com/8112138/225186616-ec6dd35e-eb4a-43ec-8acd-eae20535cbdd.mov

### After

https://user-images.githubusercontent.com/8112138/225186710-6b56a481-a2f3-4746-b192-2189848d19b7.mov

## Manual Testing Steps
- Go to latest build of Storybook in this PR
- Search `Box` in search panel
- See updates story and docs

- Pull branch and check `box.stories.tsx` and `box.d.ts` 

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
